ToNote/Controls/DragHandler.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace ToNote.Controls
+{
+    using System.Collections.ObjectModel;
+    using System.Windows;
+    using System.Windows.Controls;
+    using System.Windows.Input;
+    using System.Windows.Media;
+    using ToNote.Models;
+    using ToNote.Views;
+
+    public static class DragHandler
+    {
+        public static void HandleMouseLeftButtonDown(object sender, MouseButtonEventArgs args)
+        {
+            if (args.LeftButton == MouseButtonState.Pressed)
+            {
+                var data = new DataObject();
+                var todoControl = FindAncestorOfType<TodoControl>(sender as DependencyObject);
+                if (todoControl != null)
```

Comment:
`data.SetData("Source", todoControl ?? sender as DependencyObject);` instead of the if/else